apis/logging/v2beta/cluster_log_forwarder_types.go

```diff
@@ -0,0 +1,384 @@
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// Inputs maps input name to condition of the input.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Input Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:inputConditions"}
+	Inputs map[string]metav1.Condition `json:"inputs,omitempty"`
+
+	// Outputs maps output name to condition of the output.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Output Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:outputConditions"}
+	Outputs map[string]metav1.Condition `json:"outputs,omitempty"`
+
+	// Filters maps filter name to condition of the filter.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Filter Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:filterConditions"}
+	Filters map[string]metav1.Condition `json:"filters,omitempty"`
+
+	// Pipelines maps pipeline name to condition of the pipeline.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Pipeline Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:pipelineConditions"}
+	Pipelines map[string]metav1.Condition `json:"pipelines,omitempty"`
+}
+
+// InputSpec defines a selector of log messages for a given log type. The input is rejected
+// if more than one of the following subfields are defined: application, infrastructure, audit, and receiver.
+type InputSpec struct {
+	// Name used to refer to the input of a `pipeline`.
+	//
+	// +kubebuilder:validation:minLength:=1
+	// +required
+	Name string `json:"name"`
+
+	// NOTE: the following fields in this struct are deliberately _not_ `omitempty`.
+	// An empty field means enable that input type with no filter.
+
+	// Application, if present, enables named set of `application` logs that
+	// can specify a set of match criteria
+	//
+	// +optional
+	Application *Application `json:"application,omitempty"`
+
+	// Infrastructure, if present, enables `infrastructure` logs.
+	//
+	// +optional
+	Infrastructure *Infrastructure `json:"infrastructure,omitempty"`
+
+	// Audit, if present, enables `audit` logs.
+	//
+	//+optional
+	Audit *Audit `json:"audit,omitempty"`
+
+	// Receiver to receive logs from non-cluster sources.
+	//
+	//+optional
+	Receiver *ReceiverSpec `json:"receiver,omitempty"`
+
+	// Tuning parameters for the input.
+	//
+	//+optional
+	Tuning *InputTuningSpec `json:"tuning,omitempty"`
+}
+
+// InputTuningSpec provides tuning parameters for an input.
+type InputTuningSpec struct {
+	// RateLimitByContainer maps container names to a rate limit in records-per-second for containers
+	// with that name.
+	RateLimitByContainer map[string]int `json:"rateLimitByContainer"`
+
+	// RateLimitByNodePath maps node log file path names to a rate limit in records-per-second-per node
+	// for node logs from that path.
+	RateLimitByNodePath map[string]int `json:"rateLimitByNodePath"`
```

Comment:
Agreed, and agreed we can drop it initially if we don't come up with a clean solution. I started to think about node logs and limiting and realized we also need some way to control them, but it can wait for a later version - it's a new feature.